R/valleys.R

```diff
@@ -0,0 +1,49 @@
+#' @title Find Peak Valleys
+#'
+#' @description
+#' This function finds the valleys around peaks.
+#'
+#' @param x `numeric`, e.g. intensity values.
+#'
+#' @param p `numeric`, indices of identified peaks/local maxima.
+#'
+#' @return A `matrix` with three columns representing the index of the left
+#' valley, the peak center, and the right valley.
+#'
+#' @author Sebastian Gibb
+#' @export
+#' @examples
+#' ints <- c(5, 8, 12, 7, 4, 9, 15, 16, 11, 8, 3, 2, 3, 2, 9, 12, 14, 13, 8, 3)
+#' mzs <- seq_along(ints)
+#' peaks <- which(localMaxima(ints, hws = 3))
+#' cols <- seq_len(peaks) + 1
+#'
+#' plot(mzs, ints, type = "h", ylim = c(0, 16))
+#' points(mzs[peaks], ints[peaks], col = cols, pch = 20)
+#'
+#' v <- valleys(ints, peaks)
+#' segments(mzs[v[, "left"]], 0, mzs[v[, "right"]], col = cols, lwd = 2)
+valleys <- function(x, p) {
+    if (!is.numeric(x))
```

Comment:
Why not putting the `!is.numeric(x)` and `!is.numeric(y)` into the same `if`?

Reply:
I make `p` an `integer` here, as well. So both test a kept (now testing for `is.numeric(x)` and `is.integer(p)`).